packages/console/src/components/Tooltip/index.tsx

```diff
@@ -0,0 +1,83 @@
+import { Nullable } from '@silverhand/essentials';
+import classNames from 'classnames';
+import React, { ReactNode, RefObject, useEffect, useState } from 'react';
+import { createPortal } from 'react-dom';
+
+import * as styles from './index.module.scss';
+
+type Props<T> = {
+  content: ReactNode;
+  domRef: RefObject<Nullable<T>>;
+  className?: string;
+};
+
+type Position = {
+  top: number;
+  left: number;
+};
+
+const Tooltip = <T extends Element>({ content, domRef, className }: Props<T>) => {
+  const [tooltipDom, setTooltipDom] = useState<HTMLDivElement>();
+  const [position, setPosition] = useState<Position>();
+  const isVisible = position !== undefined;
+
+  useEffect(() => {
+    if (!domRef.current) {
+      return;
+    }
+
+    const dom = domRef.current;
+
+    const enterHandler = () => {
+      if (domRef.current) {
+        const { top, left, width } = domRef.current.getBoundingClientRect();
+        const { scrollTop, scrollLeft } = document.documentElement;
+        setPosition({ top: top + scrollTop - 12, left: left + scrollLeft + width / 2 });
```

Comment:
nit: incase of any fadeIn&Out effect, would controlling the classNames + transform + onTransitionEnd event better?

Reply:
I don't follow this one. We'll likely use react-spring for animations so it'll be good?